gh-api.el

```diff
@@ -158,6 +158,7 @@
       (let ((req (oref resp :-req)))
         (oset resp :data-received nil)
         (oset req :url next)
+        (oset req :query nil)
```

Comment:
Here we already get a fully-formed URL from the link header. We do *not* want to apply any of our own query args to it from our first request. So we null out `:query` here to just use the Link that GitHub gave us.